subsys/bluetooth/mesh/main.c

```diff
@@ -370,10 +355,17 @@ int bt_mesh_start(void)
 		bt_mesh_beacon_disable();
 	}
 
-	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
-	    bt_mesh_gatt_proxy_get() != BT_MESH_GATT_PROXY_NOT_SUPPORTED) {
-		bt_mesh_proxy_gatt_enable();
-		bt_mesh_adv_update();
+	if (!IS_ENABLED(CONFIG_BT_MESH_PROV) || !bt_mesh_prov_active() ||
+	    bt_mesh_prov_link.bearer->type == BT_MESH_PROV_ADV) {
+		if (IS_ENABLED(CONFIG_BT_MESH_PB_GATT)) {
+			(void)bt_mesh_pb_gatt_disable();
+		}
+
+		if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
+		    bt_mesh_gatt_proxy_get() != BT_MESH_GATT_PROXY_NOT_SUPPORTED) {
```

Comment:
Minor: `bt_mesh_gatt_proxy_get` already checks for `IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)`. So `if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY))` should be enough.